modules/file_upload/cms_file_upload.e

```diff
@@ -1,248 +0,0 @@
-			-- List of permission ids, used by this module, and declared.
-		do
-			Result := Precursor
-			Result.force ("manage admin")
-			Result.force ("admin users")
-			Result.force ("admin roles")
-			Result.force ("admin modules")
-			Result.force ("install modules")
-			Result.force ("admin core caches")
-			Result.force ("clear blocks cache")
-			Result.force ("admin export")
-			Result.force ("export core")
-
-			Result.force ("upload files")
-		end
-
-feature {CMS_API} -- Module Initialization
-
-	initialize (api: CMS_API)
-			-- <Precursor>
-		do
-			Precursor (api)
-		end
-
-feature {CMS_API}-- Module management
-
-	install (api: CMS_API)
-			-- install the module
-		local
-			sql: STRING
-		do
-			-- create a database table
-			if attached {CMS_STORAGE_SQL_I} api.storage as l_sql_storage then
-				if not l_sql_storage.sql_table_exists ("file_upload_table") then
-					sql := "[
-CREATE TABLE file_upload_table(
-  `id` INTEGER PRIMARY KEY AUTO_INCREMENT NOT NULL CHECK("id">=0),
-  `name` VARCHAR(100) NOT NULL,
-  `uploaded_date` DATE,
-  `size` INTEGER
-);
-					]"
-					l_sql_storage.sql_execute_script (sql, Void)
-					if l_sql_storage.has_error then
-						api.logger.put_error ("Could not initialize database for file uploader module", generating_type)
-					end
-				end
-				Precursor {CMS_MODULE}(api)
-			end
-		end
-
-feature -- Access: router
-
-
-	setup_router (a_router: WSF_ROUTER; a_api: CMS_API)
-			-- <Precursor>
-		local
-			www: WSF_FILE_SYSTEM_HANDLER
-		do
-
-			map_uri_template_agent (a_router, "/upload/", agent execute_upload, void)
-
-			create www.make_with_path (document_root)
-			www.set_directory_index (<<"index.html">>)
-			www.set_not_found_handler (agent execute_not_found_handler)
-			a_router.handle("", www, a_router.methods_get)
-		end
-
-feature -- Hooks
-
-	setup_hooks (a_hooks: CMS_HOOK_CORE_MANAGER)
-		do
-			a_hooks.subscribe_to_menu_system_alter_hook (Current)
-			a_hooks.subscribe_to_block_hook (Current)
-		end
-
-	block_list: ITERABLE [like {CMS_BLOCK}.name]
-		do
-			Result := <<"?uploader">>
-		end
-
-	get_block_view (a_block_id: READABLE_STRING_8; a_response: CMS_RESPONSE)
-		local
-			menu: CMS_MENU
-			menu_block: CMS_MENU_BLOCK
-			menu_entries_count: INTEGER
-		do
-			if a_block_id.same_string ("uploader") then
-				-- create menu_block.make ({CMS_MENU_SYSTEM}.primary_menu)
-				-- a_response.add_block (menu_block, "page_top")
-			end
-		end
-
-	menu_system_alter (a_menu_system: CMS_MENU_SYSTEM; a_response: CMS_RESPONSE)
-		local
-			link: CMS_LOCAL_LINK
-			second_link: CMS_LOCAL_LINK
-		do
-			-- login in demo did somehow not work
-			-- if a_response.has_permission ("upload files") then
-				create link.make ("Upload", "upload/")
-				a_menu_system.primary_menu.extend (link)
-			-- end
-		end
-
-feature -- Configuration		
-
-	document_root: PATH
-			-- Document root to look for files or directories
-		once
-			Result := execution_environment.current_working_path.extended ("site")
```

Comment:
better use the CMS_API.files_location .. since it can be changed by configuration